Included benchmark tests (#20)
ros2/libyaml_vendor: libyaml_q_declaration.md

The `libyaml` meets the basic requirements for a software platform in terms of testing its basic functionality, providing a [valid license](https://github.com/yaml/libyaml/blob/master/License) for the code used and a public Github repository with the changes made to the code over time.

> Even if the library does not provide an API/ABI policy targeting the desired use of the library, the fact that it deals with the YAML standard and this one hasn’t changed since 2009, allows us to infer that the functionality needed for the ROS core from this library is not going to be changed.  The imported version of `libyaml` through the `libyaml_vendor` package is fixed to the 0.18 version. This will make the API/ABI stable.

Reviewer: ```suggestion
Even though the library does not provide an API/ABI policy targeting the desired use of the library, the imported version of `libyaml` through the `libyaml_vendor` package is fixed to the 0.18 version. This will make the API/ABI stable.
```

The [safe_yaml](https://rubygems.org/gems/safe_yaml) ruby gem has over 80 million downloads and one of its implementations uses `libyaml` through psych.
It is also used in the [Go-yaml project](https://github.com/go-yaml/yaml), the project supporting YAML in the Go language.

> ROS 2 uses this library to load parameters from a file when the arguments are propely used. There is no oficial tests or code coverage information about this library. But in this case we are only interested in the portion of public API that read YAML files and we will cover these

Reviewer: ```suggestion
ROS 2 uses this library to load parameters from a file when the arguments are properly used. There are no official tests or code coverage information about this library, but in this case we are only interested in the portion of public API that read YAML files and we will cover these
```

tests and coverage in the vendored package. Including specific tests for the `libyaml` version imported by `libyaml_vendor` for the tier 1 platforms listed in [REP-2000](https://www.ros.org/reps/rep-2000.html#support-tiers).

> Considering the previously mentioned reasons, we consider this library to be robust and reliable. In terms of ROS2 package metrics to be Quality Level 1.

Reviewer: ```suggestion
Considering the previously mentioned reasons, we consider this library to be robust, reliable and at Quality Level 1.
```

# Comparison with ROS packages quality standards

## Version policy [1]

### Version Scheme [1.i]

It is not stated if the library supports any kind of version policy. But the version increases with a new release. The imported version of
`libyaml` through the `libyaml_vendor` package is fixed to the 0.18 version.

### Version Stability [1.ii]

Current version of `libyaml` in [its repository](https://github.com/yaml/libyaml) is 0.24, however, for the purposes of ROS2 Quality Level analysis, the imported version of `libyaml` through the `libyaml_vendor` package is fixed to the 0.18 version.

### Public API Declaration [1.iii]

> As a C library, elements available in its [yaml.h](https://github.com/yaml/libyaml/blob/master/include/yaml.h) are considered to be their public API.

Reviewer: ```suggestion
As a C library, elements available in [yaml.h](https://github.com/yaml/libyaml/blob/master/include/yaml.h) are considered to be the library's public API.
```

## Platform Support [6]

> This library does not state support for any specific platform. But this is build in the ROS 2 buildfarm for atll Tier 1 platforms:

Reviewer: ```suggestion
This library does not state support for any specific platform, but it is built in the ROS 2 buildfarm for all tier 1 platforms:
```

Currently nightly results can be seen here:
* [linux-aarch64_release](https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_release/lastBuild/testReport/libyaml_vendor/)
* [linux_release](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/libyaml_vendor/)
* [mac_osx_release](https://ci.ros2.org/view/nightly/job/nightly_osx_release/lastBuild/testReport/libyaml_vendor/)
* [windows_release](https://ci.ros2.org/view/nightly/job/nightly_win_rel/lastBuild/testReport/libyaml_vendor/)

## Security [7]

### Vulnerability Disclosure Policy [7.i]

> This package does not yet have a Vulnerability Disclosure Policy. But it's defined in the vendored Quality Declaration.

Reviewer 2: These 2 sentences seem to be contradicting each other. How about something along the lines of "The libyaml library does not yet have a Vulnerability Disclosure Policy, but for issues affecting ROS, see the policy defined in the vendor package."
